src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10949,175 +10946,133 @@ private ExpressionSyntax ParseSubExpressionCore(Precedence precedence)
+            //    This allows for things like assignments/lambdas in the branches of the conditional.
+            if (this.CurrentToken.Kind == SyntaxKind.QuestionToken && precedence <= Precedence.Conditional)
+                return consumeConditionalExpression(currentExpression);
 
-                var newPrecedence = GetPrecedence(opKind);
+            return currentExpression;
 
-                // check for >>, >>=, >>> or >>>=
-                int tokensToCombine = 1;
-                if (tk == SyntaxKind.GreaterThanToken
-                    && this.PeekToken(1).Kind is SyntaxKind.GreaterThanToken or SyntaxKind.GreaterThanEqualsToken
-                    && NoTriviaBetween(this.CurrentToken, this.PeekToken(1))) // check to see if they really are adjacent
-                {
-                    if (this.PeekToken(1).Kind == SyntaxKind.GreaterThanToken)
-                    {
-                        if (this.PeekToken(2).Kind is SyntaxKind.GreaterThanToken or SyntaxKind.GreaterThanEqualsToken
-                            && NoTriviaBetween(this.PeekToken(1), this.PeekToken(2))) // check to see if they really are adjacent
-                        {
-                            if (this.PeekToken(2).Kind == SyntaxKind.GreaterThanToken)
-                            {
-                                opKind = SyntaxFacts.GetBinaryExpression(SyntaxKind.GreaterThanGreaterThanGreaterThanToken);
-                            }
-                            else
-                            {
-                                opKind = SyntaxFacts.GetAssignmentExpression(SyntaxKind.GreaterThanGreaterThanGreaterThanEqualsToken);
-                                isAssignmentOperator = true;
-                            }
+            ExpressionSyntax? tryExpandExpression(ExpressionSyntax leftOperand, Precedence precedence)
+            {
+                // Look for operators that can follow what we've seen so far, and which are acceptable at this
+                // precedence level.  Examples include binary operator, assignment operators, range operators `..`, as
+                // well as `switch` and `with` clauses.
 
-                            tokensToCombine = 3;
-                        }
-                        else
-                        {
-                            opKind = SyntaxFacts.GetBinaryExpression(SyntaxKind.GreaterThanGreaterThanToken);
-                            tokensToCombine = 2;
-                        }
-                    }
-                    else
-                    {
-                        opKind = SyntaxFacts.GetAssignmentExpression(SyntaxKind.GreaterThanGreaterThanEqualsToken);
-                        isAssignmentOperator = true;
-                        tokensToCombine = 2;
-                    }
+                var (operatorTokenKind, operatorExpressionKind) = getOperatorTokenAndExpressionKind();
```

Comment:
>operatorTokenKind, operatorExpressionKind [](http://example.com/codeflow?start=21&length=41)

Consider calling these `opToken` and `opKind` to reduce the differences.

Reply:
i intentionally changed as the old names were more confising.  opToken was not a token, but was a kind.  opKind was not the kind of the op token itself, but of the op expression.  

The new names are much clearer (both are kinds) and indicate which domain they're describing (the token domain or the expression domain).